src/vector/vraster.cpp

```diff
@@ -348,37 +369,35 @@ struct VRleTask {
 
     void operator()(FTOutline &outRef, SW_FT_Stroker &stroker)
     {
-        if (mGenerateStroke) {  // Stroke Task
```

Comment:
instead of checking the size everywhere we can just bail out if the size bigger than SHORT_MAX at start of the function and after SW_FT_Stroker_GetCounts(stroker, &points, &contors) function.

NOTE: Usually path size will not be that big. if its big then certainly there is some logic problem (today i fixed one that generated by Dasher).

Reply:
By "checking everywhere" do you mean the added `assert`-s? I added them not for real world checking, but for debug mode checking for programming errors. I thought it is a good idea to `assert` that we must not signed-overflow in the place where we do that, so that if this methods are used from some other places, it could be caught. But if this is the case I can remove the asserts.

If I understand correctly this can happen also in case of such way specially prepared input files? Or the code is not supposed to work reliably in case of untrusted input files?

Reply:
I don't mind adding assert().
My point was FTOutline() is an helper internal object and it can only handle points < SHORt MAX. So before passing the path we can validate the path (whether it has points more that FToutline can handle) and return early.

ex:
 if (path.points().size > SHORT_MAX)
    return;
else
    ftOutline.convert(path);

In that way we can validate the inputs in one place and then can safely use it in Ftoutline assuming that the input data is proper.

Reply:
I see.

My point was that as an object FTOutline should know better, what data it can or can't handle, so it performs the check, not the caller. Also if you decide to use the object in some other part of the code, the checks would still be there, no need to copy them from the previous place of usage. You can even put `[[nodiscard]]` on returned `bool` so that the caller should do something with the fact that he has invalid data.

But if it is supposed to be used in one place only I can add the checks above the usage and remove them from `grow` and `convert`. Right now all the checks are only in the `grow` method, so in one place as well.